api/configuration/v1/kongconsumer_types.go

```diff
@@ -34,6 +34,8 @@ import (
 // +kubebuilder:printcolumn:name="Programmed",type=string,JSONPath=`.status.conditions[?(@.type=="Programmed")].status`
 // +kubebuilder:validation:XValidation:rule="has(self.username) || has(self.custom_id)", message="Need to provide either username or custom_id"
 // +kubebuilder:validation:XValidation:rule="!has(oldSelf.spec.controlPlaneRef) || has(self.spec.controlPlaneRef)", message="controlPlaneRef is required once set"
+//  REVIEW: allow same namespace?
+// +kubebuilder:validation:XValidation:rule="!has(self.spec.controlPlaneRef.konnectNamespacedRef) ? true : !has(self.spec.controlPlaneRef.konnectNamespacedRef.__namespace__)", message="spec.controlPlaneRef cannot specify namespace for namespaced resource"
```

Comment:
Why `__namespace__` and not `namespace`? 🤔

Reply:
Because `namespace` is a keyword in CEL rules. You HAVE TO use `__namespace__` to refer to fields with name `namespace`, otherwise you will receive a `compilation failed: ERROR: <input>:1:9: undefined field 'namespace'` error in installing the CRD: https://github.com/kubernetes/kubernetes/issues/122777.

Reply:
TIL. 👍